dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.PushNotifications
+{
+    [experimental]
+    // Event args for the Push payload.
+    runtimeclass PushNotificationReceivedEventArgs
+    {
+        // Initialize using the IBackgroundInstance: used specifically for the Background Activation scenario
+        static PushNotificationReceivedEventArgs CreateFromBackgroundTaskInstance(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+        // Initialize using the PushNotificationEventArgs from Windows: used specifically for in-memory event handling when app is already in foreground
+        static PushNotificationReceivedEventArgs CreateFromPushNotificationReceivedEventArgs(Windows.Networking.PushNotifications.PushNotificationReceivedEventArgs args);
+
+        // The Push payload
+        byte[] Payload { get; };
+
+        // Gets a deferral to run under specific modes like low power mode
+        Windows.ApplicationModel.Background.BackgroundTaskDeferral GetDeferral();
+
+        // Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+        event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+        // Set to true to prevent proceeding launch due to Background Activation: false by default
+        Boolean Handled;
+    };
+
+    [flags]
+    enum PushNotificationRegistrationOption
+    {
+        PushTrigger = 0x1, // Registers a Push Trigger with Background Infrastructure
+        ComActivator = 0x2, // Registers the Project Reunion Background Task component as an InProc COM server 
+    };
+
+    [experimental]
+    runtimeclass PushNotificationActivationInfo
+    {
+        // Initialize using a RegistrationOption and optionally defined parameters like manifest defined activatorId
+        // 1) If option = PushTrigger is specified, only the Push Trigger will be Registered with Background Infra
```

Comment:
What is "Background Infra"? There are no hits on docs.microsoft.com. There s one mention buried deep in the hardware documentation of a "[Background Infrastructure](https://docs.microsoft.com/en-us/windows-hardware/design/device-experiences/activators)". Is that what "Background Infra" is?

This is a public repo. Please use terms that people outside Microsoft will understand. Can we avoid having to introduce new terminology for the purpose of a comment?

Maybe "... will be registered with the background task infrastructure"?

Reply:
It is "Background Infrastructure", no need to save chars here 😁.